x/observer/simulation/operations.go

```diff
@@ -0,0 +1,101 @@
+package simulation
+
+import (
+	"math/rand"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/zeta-chain/node/x/observer/keeper"
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+// Simulation operation weights constants
+// Operation weights are used by the simulation program to simulate the weight of different operations.
+// This decides what percentage of a certain type of operation is part of a block.
+// Based on the weights assigned in the cosmos sdk modules , 100 seems to the max weight used , and therefore guarantees that at least one operation of that type is present in a block.
+// TODO Add more details to comment based on what the number represents in terms of percentage of operations in a block
+// https://github.com/zeta-chain/node/issues/3100
+const (
+	// #nosec G101 not a hardcoded credential
+	OpWeightMsgTypeMsgEnableCCTX = "op_weight_msg_enable_crosschain_flags"
+	// DefaultWeightMsgTypeMsgEnableCCTX We ues a high weight for this operation
+	// to ensure that it is present in the block more number of times than any operation that changes the validator set
+
+	// Arrived at this number based on the weights used in the cosmos sdk staking module and through some trial and error
+	DefaultWeightMsgTypeMsgEnableCCTX = 3650
```

Comment:
Can we confirm adding a flag to enable or disable staking hooks?
Having the staking hooks causes our simulations to have this unrealistic number of transactions of the type `MsgEnableCCTX`

Reply:
I don't understand the relationship, you should be able to give a lower weight for a message to be used less

Reply:
We try to make sure, that there is a messageEnable CCTX to counter every action that can disable CCTX. 
Actions such as `RemoveValidator,` `SlashValidator`.
The higher number makes sure that we have more occurrences for messageEnableCCTX.

Ill try to find documenation but this is the function used to select the operation 

```
func (ops WeightedOperations) getSelectOpFn() simulation.SelectOpFn {
	totalOpWeight := ops.totalWeight()

	return func(r *rand.Rand) simulation.Operation {
		x := r.Intn(totalOpWeight)
		for i := 0; i < len(ops); i++ {
			if x <= ops[i].Weight() {
				return ops[i].Op()
			}

			x -= ops[i].Weight()
		}
		// shouldn't happen
		return ops[0].Op()
	}
}
```